client.go

```diff
@@ -1,20 +1,29 @@
-package redis
+package dpredis
 
 import (
-	"context"
 	"errors"
-	"github.com/ONSdigital/log.go/log"
 	"time"
 
+	. "github.com/ONSdigital/dp-redis/interfaces"
 	"github.com/ONSdigital/dp-sessions-api/session"
-	goredis "github.com/go-redis/redis"
+	"github.com/go-redis/redis"
 )
 
-var ctx = context.Background()
+var (
+	ErrEmptySession   = errors.New("session is empty")
+	ErrEmptyAddress   = errors.New("address is empty")
+	ErrEmptyPassword  = errors.New("password is empty")
+	ErrInvalidTTL     = errors.New("ttl should not be zero")
```

Comment:
Sorry I am being really picky now - but as these error are being returned to the calling code the context is diluted a bit - if you saw `address is empty` in the logs you wouldn't know necessarily what that was referring to. Something like:

`redis host address required but was empty`, `redis client ttl cannot be 0` etc. 

give you more context of the error scenario.